### PR TITLE
Schema name add default value

### PR DESCRIPTION
**Pull request by zjcnb**
For #11621 .

add default value

### CODE REVIEW

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/api/ShardingSphereDataSourceFactory.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ public static DataSource createDataSource(final Map<String, DataSource> dataSour
      * @throws SQLException SQL exception
      */
     public static DataSource createDataSource(final String schemaName, final DataSource dataSource, final Collection<RuleConfiguration> configs, final Properties props) throws SQLException {
-        return createDataSource(schemaName, Collections.singletonMap(schemaName, dataSource), configs, props);
+        return createDataSource(schemaName, Collections.singletonMap(Strings.isNullOrEmpty(schemaName) ? DefaultSchema.LOGIC_NAME : schemaName, dataSource), configs, props);
```
Comment (terrymanu) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/api/ShardingSphereDataSourceFactory.java:79]: The inline method `createDataSource` has already check null of schemaName, is it necessary to add check again?

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/resources/META-INF/namespace/datasource.xsd [hunk 2]
```diff
@@ -29,7 +29,7 @@
                 <xsd:element ref="beans:props" minOccurs="0" />
             </xsd:all>
             <xsd:attribute name="id" type="xsd:string" use="required" />
-            <xsd:attribute name="schema-name" type="xsd:string" />
+            <xsd:attribute name="schema-name" type="xsd:string" default="logic_db"/>
```
Comment (terrymanu) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/resources/META-INF/namespace/datasource.xsd:32]: Please keep one space between `"logic_db"` and `/>`, just keep consist with others

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java [hunk 3]
```diff
@@ -40,6 +42,10 @@
      */
     public static String getSchemaName(final Environment environment) {
         StandardEnvironment standardEnv = (StandardEnvironment) environment;
-        return standardEnv.getProperty(PREFIX + SCHEMA_NAME);
+        String schemaName = standardEnv.getProperty(PREFIX + SCHEMA_NAME);
+        if (StringUtils.isEmpty(schemaName)) {
+            return DefaultSchema.LOGIC_NAME;
+        }
+        return schemaName;
```
Comment (terrymanu) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java:49]: The return value should name as `result`

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java [hunk 4]
```diff
@@ -40,6 +42,10 @@
      */
     public static String getSchemaName(final Environment environment) {
         StandardEnvironment standardEnv = (StandardEnvironment) environment;
-        return standardEnv.getProperty(PREFIX + SCHEMA_NAME);
+        String schemaName = standardEnv.getProperty(PREFIX + SCHEMA_NAME);
+        if (StringUtils.isEmpty(schemaName)) {
```
Comment (terrymanu) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java:46]: Please use `Strings`, just keep consist with other usage

Comment (terrymanu) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java:46]: Is it passable to use ternary operator?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/api/ShardingSphereDataSourceFactory.java (hunk 1 above)
Comment (zjcnb) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/api/ShardingSphereDataSourceFactory.java:79]: it 's only check schemaName, but not check datasourceMap key

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/resources/META-INF/namespace/datasource.xsd (hunk 2 above)
Comment (zjcnb) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/resources/META-INF/namespace/datasource.xsd:32]: ok, i will done

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java (hunk 4 above)
Comment (zjcnb) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java:46]: > Is it passable to use ternary operator?

yes

**Review by terrymanu — APPROVED** [review]

**Pull request merged**